src/e2e-test/java/io/cdap/plugin/common/ValidationHelper.java

```diff
@@ -0,0 +1,271 @@
+        return false;
+      }
+    }
+    return true;
+  }
+
+  public static boolean validateJsonElement(JsonElement v1, JsonElement v2) {
+    if (v1.isJsonObject()) {
+      if (!v2.isJsonObject()) {
+        return false;
+      }
+      return validateJsonObject(v1.getAsJsonObject(), v2.getAsJsonObject());
+
+    } else if (v1.isJsonArray()) {
+      if (!v2.isJsonArray()) {
+        return false;
+      }
+      return validateJsonArray(v1.getAsJsonArray(), v2.getAsJsonArray());
+
+    } else if (v1.isJsonPrimitive()) {
+      if (v2.isJsonPrimitive()) {
+        return validateJsonPrimitive(v1.getAsJsonPrimitive(), v2.getAsJsonPrimitive());
+      } else if (v2.isJsonArray()) {
+        return validateByteArray(v1.getAsJsonPrimitive(), v2.getAsJsonArray());
+      }
+    } else if (v1.isJsonNull()) {
+      return v2.isJsonNull();
+    }
+
+    return false;
+  }
+
+  public static boolean validateJsonObject(JsonObject v1, JsonObject v2) {
+    for (Map.Entry<String, JsonElement> entry : v1.entrySet()) {
+      String key = entry.getKey();
+      JsonElement value1 = entry.getValue();
+      JsonElement value2 = v2.get(key);
+
+      if (!validateJsonElement(value1, value2)) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  public static boolean validateJsonPrimitive(JsonPrimitive v1, JsonPrimitive v2) {
+    if (!v1.isJsonPrimitive() || !v2.isJsonPrimitive()) {
+      return false;
+    }
+
+    JsonPrimitive p1 = v1.getAsJsonPrimitive();
+    JsonPrimitive p2 = v2.getAsJsonPrimitive();
+
+    if (p1.isString() && p2.isNumber()) {
+      System.out.println("String number comparison" + p1 + p2);
+      String s1 = p1.getAsString();
+      long s2 = p2.getAsLong();
+
+      try {
+        LocalDate date = LocalDate.parse(s1);
+        long epochDay = date.toEpochDay();
+        if (epochDay == s2) {
+          return true;
+        }
+      } catch (DateTimeParseException e3) {
+        try {
+          LocalTime time1 = LocalTime.parse(s1, DateTimeFormatter.ofPattern("HH:mm:ss"));
+          long seconds1 = time1.truncatedTo(ChronoUnit.SECONDS).toSecondOfDay();
+          LocalTime time = LocalTime.ofSecondOfDay(seconds1);
+          String formattedTime = time.format(DateTimeFormatter.ofPattern("HH:mm:ss"));
+          if (formattedTime.equals(s1)) {
+            return true;
+          }
+        } catch (DateTimeParseException e1) {
+          // Date parsing failed, continue with time parsing
+
+          try {
+            LocalDateTime dateTime1 = LocalDateTime.parse(s1, DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss'Z'"));
+            long epochSeconds1 = dateTime1.toEpochSecond(ZoneOffset.UTC) * 1_000_000L;
+
+            if (epochSeconds1 == s2) {
+              return true;
+            }
+          } catch (DateTimeParseException e) {
+            // Date parsing failed, continue with other comparisons
+          }
+
+          return s1.equals(s2);
```

Comment:
In this scope `s1` and `s2` are of different type `(String,long)`
This statement will always be false